test/linalg/diagonal.jl

```diff
@@ -158,6 +160,14 @@ srand(1)
         @test (r = full(D) * U   ; A_mul_B!(UU, D, U) ≈ r ≈ UU)
         @test (r = full(D)' * U  ; Ac_mul_B!(UU, D, U) ≈ r ≈ UU)
         @test (r = full(D).' * U ; At_mul_B!(UU, D, U) ≈ r ≈ UU)
+
+        VV = full(D)
+        DD = copy(D)
+        @test (r = VV * full(D)   ; full(A_mul_B!(VV, DD)) ≈ r ≈ full(D)*full(D))
```

Comment:
why does the `r = VV * full(D)` need to be inside the `@test` instead of on its own line?